src/components/ItemsList.js

```diff
@@ -23,13 +32,33 @@ const ItemsList = () => {
     { idField: 'documentId' },
   );
 
-
   const markItemAsPurchased = (index) => {
     const { items, documentId } = shoppingList[0];
-
-    items[index].lastPurchasedOn = items[index].lastPurchasedOn
-      ? null
-      : Date.now();
+    const { lastPurchasedOn } = items[index];
+    const shoppingItemObject = items[index];
+    if (lastPurchasedOn === null) {
+      shoppingItemObject.lastPurchasedOn = Date.now();
+      shoppingItemObject.numberOfPurchases++;
+      shoppingItemObject.daysLeftForNextPurchase = calculateEstimate(
+        undefined,
```

Comment:
I have a feeling that 'undefined' shouldn't be passed to a function/assigned to a variable etc. as it generally means that a variable/argument has been declared but not passed any value, whereas "null" means there is no value, so I think null would be more appropriate here?

Reply:
I think that the lastPurchasedOn can never be null as when the user adds an item then the purchased date can be added and when the user marks it for re-purchase then the lastPurchasedOn date can be used.

Reply:
I would agree here with declaring null instead of undefined 👍

Reply:
Can you please be kind and explain why are we using `undefined` here? I'm not sure if I'm following the logic of why we are using `calculateEstimate` function in the first `if` statement block. 🙏 

I totally get the second else block from line 50 - 59. Thank you!

Reply:
Thanks @rb50  and @alexlsalt  for pointing that out. Actually invoking that function is not necessary at that point because it returns the same value passed. We shall have to remove it entirely.

Reply:
Thank you for the reviews, we have taken out the function call entirely